homeassistant/components/proximity.py

```diff
@@ -0,0 +1,337 @@
+					ATTR_DIR_OF_TRAVEL: 'arrived', 
+					ATTR_NEAREST_DEVICE: entity_name, ATTR_HIDDEN: False}
+                hass.states.set(ENTITY_ID, 0, entity_attributes)
+                _LOGGER.debug('%s Update entity: distance = 0: direction = '
+					'arrived: device = %s', ENTITY_ID, entity_name)
+            else:
+                _LOGGER.debug('%s Entity not updted: %s is in proximity '
+					'zone:%s', ENTITY_ID, entity_name, config[DOMAIN]['zone'])
+            return
+
+        """========================================================"""
+        # check that the device is not in an ignored zone
+        if new_state.state in ignored_zones:
+            _LOGGER.info('%s Device is in an ignored zone: %s', ENTITY_ID, 
+				device)
+            return
+
+        """========================================================"""
+        # check for latitude and longitude (on startup these values may not
+		# exist)
+        if not 'latitude' in new_state.attributes:
+            _LOGGER.info('%s: not LAT or LONG current position cannot be '
+				'calculated', entity_name)
+            return
+
+        # calculate the distance of the device from the monitored zone
+        distance_from_zone = round(distance(proximity_latitude,
+			proximity_longitude, new_state.attributes['latitude'],
+			new_state.attributes['longitude']) / 1000, 1)
+        _LOGGER.info('%s: distance from zone is: %s km', entity_name, 
+			distance_from_zone)
+
+        """========================================================"""
+        # compare distance with other devices
+        # default behaviour
+        device_is_closest_to_zone = True
+
+        for device in proximity_devices:
+            # ignore the device we're working on
+            if device == entity:
+                continue
+
+            # get the device state
+            device_state = hass.states.get(device)
+            if device_state in ignored_zones:
+                _LOGGER.debug('%s: no need to compare with %s - device is in '
+					'ignored zone', entity_name, device)
+                continue
+
+            # check that the distance from the proximity zone can be calculated
+            if not 'latitude' in device_state.attributes:
+                _LOGGER.debug('%s: cannot compare with %s - no location '
+					'attributes', entity_name, device)
+                continue
+
+            # calculate the distance from the proximity zone for the compare
+			# device
+            compare_distance_from_zone = round(distance(proximity_latitude,
+				proximity_longitude, device_state.attributes['latitude'],
+				device_state.attributes['longitude'])/1000, 1)
+            _LOGGER.debug('%s: compare device %s: co-ordintes: LAT %s: LONG: '
+				'%s', entity_name, device, device_state.attributes['latitude'],
+				device_state.attributes['longitude'])
+
+            # compare the distances from the proximity zone
+            if distance_from_zone < compare_distance_from_zone:
```

Comment:
These 3 if statements hold no purpose. You can rewrite it as this and save like 15 lines of code.
```python
if distance_from_zone >= compare_distance_from_zone:
    device_is_closest_to_zone = False
    break
```